avaframe/com1DFA/particleInitialisation.py

```diff
@@ -0,0 +1,248 @@
+    """ Redistribute particles so that SPH force reduces with fixed particles as boundaries
+
+        Parameters
+        ------------
+        cfg: configparser object
+            configuration settings
+        particles: dict
+            dictionary with particle properties
+        dem: dict
+            dictionary with dem header and data
+        fields: dict
+            dictionary with fields of result types
+        inputSimLines: dict
+            info on input files
+        relThField: numpy array
+            release thickness field
+
+        Returns
+        --------
+        particles: dict
+            updated particles dict with new positions
+        fields: fields
+            updated fields dictionary
+        """
+
+    particles['iterate'] = True
+    particlesList = [particles.copy()]
+    countIterations = 0
+    # load relRaster from buffered release line
+    relRaster = inputSimLines['releaseLineBuffer']['rasterData']
+
+    # redistribute particles to reduce SPH force
+    while particles['iterate'] and countIterations < cfg['GENERAL'].getint('maxIterations'):
+
+        # compute artificial viscosity effect on velocity
+        particles, force = DFAfunC.computeIniMovement(cfg['GENERAL'], particles, dem, cfg['GENERAL'].getfloat('dtIni'),
+            fields)
+
+        # compute SPH force
+        particles, force = DFAfunC.computeForceSPHC(cfg['GENERAL'], particles, force, dem,
+            cfg['GENERAL'].getint('sphOptionIni'), gradient=0)
+
+        # update position as a result of SPH force and artifical viscosity
+        particles = DFAfunC.updatePositionC(cfg['GENERAL'], particles, dem, force, cfg['GENERAL'].getfloat('dtIni'),
+            typeStop=1)
+
+        particlesList.append(particles.copy())
+
+        # compute neighbours and update fields
+        particles = DFAfunC.getNeighborsC(particles, dem)
+        particles, fields = DFAfunC.updateFieldsC(cfg['GENERAL'], particles, dem, fields)
+
+        # count iterations
+        countIterations = countIterations + 1
+
+    # reset iterate for performing ava simulations
+    particles['iterate'] = True
+
+    # remove particles that lie outside of release polygons
+    if len(relThField) != 0:
+        relRaster = relThField
+    particles = resetMassPerParticle(cfg, particles, dem, relRaster)
+    particles = com1DFA.checkParticlesInRelease(particles, inputSimLines['releaseLine'],
+            cfg['GENERAL'].getfloat('thresholdPointInPoly'))
+
+    # adjust mass of particles in order to match good final mass
+    if len(relThField) != 0:
+        particles['m'] = particles['mIni']
+        particles['mTot'] = np.sum(particles['m'])
+    else:
+        newMass = np.sum(particles['mIni'])
+        oldMass = np.sum(particles['m'])
+        particles['m'] = particles['m'] * (newMass / oldMass)
+        particles['mTot'] = np.sum(particles['m'])
+        log.info('oldMass: %.2f and newMass: %.2f - mass factor: %.2f - total mass: %.2f' %
```

Comment:
I would say this also needs to drop to debug

Reply:
done